src/components/FileInput.js

```diff
@@ -85,6 +100,7 @@ class FileInput extends Component {
         if (onChangeCallback) {
           onChangeCallback(this.state);
         }
+        this.getImageThumbnail(file);
```

Comment:
The `getImageThumnail` could go in the upper block (when we get the image). Right now we call `setState` to time, to set the file and in the callback of `setState`. Since there's no necessity for that we can avoid and extra re render.